### ISSUE TITLE
q values for combos of format and profile

### ISSUE THREAD

**Issue by ericprud** [ProfileNegotiation/I-D-Accept--Schema#20]
You may want to tie q values to combos of format and profile. You could do that in a media type header like `Accept`:
```
Accept:
    text/format-1;profile="profile-A";q=1.0,
    text/format-2;profile="profile-A";q=0.9,
    text/format-2;profile="profile-B";q=0.8,
    text/format-1;profile="profile-B";q=0.7
```
I originally thought FHIR had a use case but now believe that `Accept-Profile:` is more appropriate 'cause the profile q values are orthogonal to the format  values.

**Comment by RubenVerborgh** [comment]
We could _optionally_ do that, I think. I wouldn't make it mandatory because then you essentially mandate combinatorial explosion.

**Comment by ericprud** [comment]
Right, this would be for when your matrix of requirements demands it. In a minimalist sense, this doesn't require any change to HTTP headers as one is free to inject whatever parameters one wants into the accept types. If we raise the bar and say that the `profile` parameter is somehow equivalent to the arguments of an `Accept-Profile` (e.g. for more complex calculations of the optimum response representation), we'd have to reserve some behavior for the media type parameter `profile`. OTOH, this sort of behavior could always be added at a later date.

**Comment by larsgsvensson** [comment]
> this doesn't require any change to HTTP headers as one is free to inject whatever parameters one wants into the accept types.

IIRC the use of the `profile` parameter in an `Accept`-Header requires that the media type supports that. Of the ones registered with IANA only `application/xhtml+xml` and `application/ld+json` support profiles.

So in most cases we would need to have separate q-values for media types and profiles.

**Comment by ericprud** [comment]
> So in most cases we would need to have separate q-values for media types and profiles.

or update the registrations after `Accept-Profile` established some common language for this meaning of a profile.
